### PR TITLE
User secret management

### PR DESCRIPTION
**Pull request by bsatoriu**
Add support for creating and retrieving user secrets within MAAP for use inside algorithms. See ticket: https://github.com/MAAP-Project/Community/issues/1048

Changes:
- Add secrets endpoints that use the `cryptography` library for encrypting secret values. Encrypted secrets are stored in a new [`member_secret`](https://github.com/MAAP-Project/maap-api-nasa/blob/feature/user-secrets-1048/api/models/member_secret.py) table.
- Refactor `api/member` namespaces to move admin-specific endpoints into a new `api/admin` namespace.

### CODE REVIEW

File: api/endpoints/admin.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+log = logging.getLogger(__name__)
+
+ns = api.namespace('admin', description='Operations related to the MAAP admin')
+
+
+@ns.route('/pre-approved')
+class PreApprovedEmails(Resource):
+
+    @api.doc(security='ApiKeyAuth')
+    @login_required
```
[the remark this answers is not in the document]
Comment (bsatoriu) [reply api/endpoints/admin.py]: I added an admin role argument to restrict this endpoint to admin users only.

File: api/endpoints/admin.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            return err_response("Valid email is required.")
+
+        pre_approved_email = db.session.query(PreApproved).filter_by(email=email).first()
+
+        if pre_approved_email is not None:
+            return err_response(msg="Email already exists")
+
+        new_email = PreApproved(email=email, creation_date=datetime.utcnow())
+
+        db.session.add(new_email)
```
[the remark this answers is not in the document]
Comment (bsatoriu) [reply api/endpoints/admin.py:72]: I added an admin role argument to restrict this endpoint to admin users only.

File: api/endpoints/admin.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        pre_approved_schema = PreApprovedSchema()
+        return json.loads(pre_approved_schema.dumps(new_email))
+
+
+@ns.route('/pre-approved/<string:email>')
+class PreApprovedEmails(Resource):
+
+    @api.doc(security='ApiKeyAuth')
+    @login_required
+    def delete(self, email):
```
[the remark this answers is not in the document]
Comment (bsatoriu) [reply api/endpoints/admin.py:84]: I added an admin role argument to restrict this endpoint to admin users only.

File: api/endpoints/members.py [hunk 4]
```diff
@@ -432,6 +426,120 @@ def delete(self):
...
+                    .filter_by(member_id=get_authorized_user().id)
+                    .order_by(MemberSecret_db.secret_name)
+                    .all())
+
+            result = [{
+                'secret_name': s.secret_name
+            } for s in secrets]
+
+            return result
+        except Exception as ex:
```
[the remark this answers is not in the document]
Comment (bsatoriu) [reply api/endpoints/members.py]: I changed this to a SqlAlchemy exception type.

**Comment by bsatoriu** [comment]
@sujen1412 the admin role restriction for pre_approved_email endpoints and improved exception handling is ready for final review.

**Review by sujen1412 — APPROVED** [review]

**Pull request merged**